src/database/adatabasesetup.cpp

```diff
@@ -302,12 +302,12 @@ bool ADataBaseSetup::backupOldData()
         return true;
     }
 
-    auto date_string = QDateTime::currentDateTime().toString(Qt::ISODate);
+    auto date_string = QDateTime::currentDateTime().toString("yyyy_MM_dd_T_hh_mm");
```

Comment:
From 20b697d14ea257b985a6292a72b519651ac5d7d1
```c++
auto date_string = QDateTime::currentDateTime().toString(Qt::ISODate);
#if defined(_WIN32) || defined(_WIN64)
date_string.replace(':', '-');
#endif
```

Reply:
better to keep the filename consistent across platforms

Reply:
Then we need our custon function to be used instead of `.toString(...)` because hardcoding it that way isnt preferable
```c++
QString dateToString(QDateTime& dt) {
    ...
    return custom_time_string
}
```

Reply:
Why add overhead by writing a custom function for `.toString()` that does the same as `.toString()` but in a non-standard and probably less efficient way?

What would that function do other than

```cpp
QString dateToString(QDateTime& dt) {
    auto custom_time_string = dt.toString("yyyy_MM_dd_T_hh_mm");
    return custom_time_string
}
```
?

If it does only that, why add the overhead? If it does something more, why would that be required?

Reply:
because we control the output to and from (if needed) in a form way. The "overhead" is negligible, the readability inscreased and we can give the compiler hint to inline.
We pack the custom logic of reading and writing the backup file string into 2 functions that hide any .toString(), str.replace()

Reply:
For dates this is no issue, since isodate does not include the time, only datetime does.
`date.toString(Qt::ISODate)` returns `yyyy-MM-dd`

For reading the backup files, checking the filename is a flawed approach, since:

1. We cannot control how the user may rename his backups.
2. The file name says nothing about the validity of its contents.

I agree that the overhead would be negligible, but I prefer the simplicity of using the function that Qt provides over duplicating its functionality and employing custom logic.

QDate and QDateTime, date and time operations are immensely complex. Since Qt has a function that does exactly what we need (output the current date and time in a format we control) - `.toString()` I prefer not to include custom logic that does little more than duplicating what is already included in Qt.

However, I do agree with you that hardcoding a string here is not the perfect solution. I have added a namespace that collects constants that are used throughout the application - this will ensure uniformity of date, time and datetime formats and adresses the isssue with this function.

See [19232fa](https://github.com/fiffty-50/openpilotlog/pull/55/commits/19232fa5981e73e0d42b39a3aabcc075ffe0c328)